src/runtime_src/core/edge/drm/zocl/common/zocl_sysfs.c

```diff
@@ -598,27 +599,29 @@ static ssize_t read_xclbin_full(struct file *filp, struct kobject *kobj,
 
 	read_lock(&zdev->attr_rwlock);
 
-	for (i = 0; i < MAX_PR_SLOT_NUM; i++) {
-		zocl_slot = zdev->pr_slot[i];
-		if (!zocl_slot || !zocl_slot->axlf)
-			continue;
+	// Only read slot 0's xclbin - TODO: extend to multi-slot 
```

Comment:
Didn't get why can't we keep the loop. Anything I am missing ?

Reply:
Because if you loop through all the slots, you'll be combining multiple xclbins when trying to access it, and it fails.
Ideally the proper way to access it is to allow passing of a slot argument to access the desired xclbin.
This solution works since we only allow 1 PL xclbin in slot 0, and that is the one we need to load through UUID.

Reply:
As per the offline discussion with Jeff, it is clear now. Approving.